File: core/identity/runtime_identity_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.identity.browser_fingerprint_engine import fingerprint_browser_identity


def build_runtime_identity(
    identity: Dict[str, Any],
) -> Dict[str, Any]:
    return {
        "runtime_identity": fingerprint_browser_identity(identity),
        "profile_id": identity.get("profile_id", "default"),
        "bounded": True,
    }

File: src/identity/runtimeIdentityEngine.ts
/**
 * Converted from Python: core/identity/runtime_identity_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { fingerprintBrowserIdentity } from "./browserFingerprintEngine.js";

export function buildRuntimeIdentity(identity: any): any {
  return {"runtime_identity": fingerprintBrowserIdentity(identity), "profile_id": py.get(identity, "profile_id", "default"), "bounded": true};
}
export { fingerprintBrowserIdentity };
